new properties (license, version) from citation.cff. Fixes #935 (#948)
KnowledgeCaptureAndDiscovery/somef: docs/index.md

  - URL: website or ORCID associated with the author
  - Affiliation: name of organization or affiliation
- **Build file**: Build file(s) of the project. For example, files used to create a Docker image for the target software, package files, etc.

> - **Citation**: Preferred citation as the authors have stated in their readme file. SOMEF recognizes Bibtex, Citation File Format files and other means by which authors cite their papers (e.g., by in-text citation). 

Reviewer: ```suggestion
- **Citation**: Preferred citation(s) as the authors have stated in their readme file. SOMEF recognizes Bibtex, Citation File Format files and other means by which authors cite their papers (e.g., by in-text citation). 
```

> For CITATION.cff files, SOMEF now generates two separate entries: one for the software (is_preferred_citation: False) and another for the preferred citation (is_preferred_citation: True). This ensures metadata like DOI or version is correctly assigned to each entity.

Reviewer: but if there is no preferred citation, then the software one is not false.
Instead of generating `False`, I would only leave `is_preferred_citaiton: True` when we find prefered citation. And do not add anything otherwise.

> We aim to recognize the following properties:

Reviewer: ```suggestion
We recognize the following properties:
```

  - Title: Title of the publication
  - Author: list of author names in the publication
  - URL: URL of the publication 
  - DOI: Digital object identifier of the publication
  - Date published

>   - Version: Software version (if applicable)

Reviewer: ```suggestion
  - Version: Software version (if applicable, i.e., the main citation is a software deposit)
```

  - URL: website or ORCID associated with the author
  - Affiliation: name of organization or affiliation
- **Build file**: Build file(s) of the project. For example, files used to create a Docker image for the target software, package files, etc.
- **Citation**: Preferred citation(s) as the authors have stated in their readme file. SOMEF recognizes Bibtex, Citation File Format files and other means by which authors cite their papers (e.g., by in-text citation). 

> For CITATION.cff files, SOMEF now generates two separate entries: one for the software and another for the preferred citation (is_preferred_citation: True). This ensures metadata like DOI or version is correctly assigned to each entity.

Reviewer: ```suggestion
For CITATION.cff files, SOMEF now generates two separate entries: one for the software tool and another for the preferred citation (if available). This ensures metadata like DOI or version is correctly assigned to each entity.
```

- **Forks url**: Links to forks made of the project
- **Full name**: Name + owner (owner/name)
- **Full title**: If the repository is a short name, we will attempt to extract the longer version of the repository name
- **Funding**: Funding information associated with the project. **Note**: Currently, this information is only extracted from existing `codemeta.json` files within the repository.
- **Homepage**: URL of the item.
- **Identifier**: Identifier associated with the software (if any), such as Digital Object Identifiers and Software Heritage identifiers (SWH). DOIs associated with publications will also be detected. 
- **Images**: Images used to illustrate the software component
- **Installation instructions**: A set of instructions that indicate how to install a target repository
- **Invocation**: Execution command(s) needed to run a scientific software component
- **Issue tracker**: Link where to open issues for the target repository
- **Keywords**: set of terms used to commonly identify a software component

> - **License**: License and usage terms of a software component. 

Reviewer: ```suggestion
- **License**: License and usage terms of a software component 
```

- **Invocation**: Execution command(s) needed to run a scientific software component
- **Issue tracker**: Link where to open issues for the target repository
- **Keywords**: set of terms used to commonly identify a software component

> - **License**: License and usage terms of a software component. Now we also extract license from citation.cff.

Reviewer: ```suggestion
- **License**: License and usage terms of a software component. 
```

- **Repository status**: Repository status as it is described in [repostatus.org](https://www.repostatus.org/).
- **Requirements**: Pre-requisites and dependencies needed to execute a software component
- **Run**: Running instructions of a software component. It may be wider than the `invocation` category, as it may include several steps and explanations.

> - **Runtime platform**: specifies the runtime environment or script interpreter dependencies (e.g., Python, Java).

Reviewer: ```suggestion
- **Runtime platform**: specifies the runtime environment or script interpreter dependencies (e.g., Python, Java)
```
